### PR TITLE
Journalist can sign in

### PR DESCRIPTION
**Pull request by amandagibson**
#### Description
Changes proposed in this pull request:

* Log In for Employees

#### What I have learned working on this feature:
We learned about devise helper methods

#### Screenshots:
Screenshot 2019-03-18 at 11 15 45
Screenshot 2019-03-18 at 11 15 57
Screenshot 2019-03-18 at 11 16 14

Screenshot 2019-03-20 at 17 25 29

### CODE REVIEW

File: features/journalist_can_sign_in.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	I would like to log in as a Journalist
+
+	Background:
+		Given the following users exist
+			| email           | role       |
+			| amanda@mail.com | journalist |
+			| viktor@mail.com | user       |
+
+	Scenario: Logging in as Journalist (access to staff page)
+		When I am logged in as "amanda@mail.com"
```
Comment (LauraRe) [inline features/journalist_can_sign_in.feature]: I would use Given here

File: features/journalist_can_sign_in.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		And I visit the "landing" page
+		And I click on "Employee"
+		Then I should see "Welcome to the staff page"
+
+	Scenario: Logging in as User (not access to staff page)
+		When I am logged in as "viktor@mail.com"
```
Comment (LauraRe) [inline features/journalist_can_sign_in.feature]: same as line 13

File: app/controllers/employees_controller.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class EmployeesController < ApplicationController
+	def index
```
Comment (LauraRe) [inline app/controllers/employees_controller.rb]: should we prevent access to the page to normal users? or can we do it later in a separate feature?

Comment (vick3d) [reply app/controllers/employees_controller.rb]: will look into this

File: db/migrate/20190318083136_add_column_to_user.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddColumnToUser < ActiveRecord::Migration[5.2]
+	def change
+		add_column :users, :role, :string
```
Comment (LauraRe) [inline db/migrate/20190318083136_add_column_to_user.rb]: user role should be an integer

File: features/journalist_can_sign_in.feature (hunk 2 above)
Comment (amandagibson) [reply features/journalist_can_sign_in.feature]: done

File: features/journalist_can_sign_in.feature (hunk 1 above)
Comment (amandagibson) [reply features/journalist_can_sign_in.feature]: done

File: db/migrate/20190318083136_add_column_to_user.rb (hunk 4 above)
Comment (amandagibson) [reply db/migrate/20190318083136_add_column_to_user.rb]: we will fix this after the stripe merge.

File: app/controllers/employees_controller.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class EmployeesController < ApplicationController
+	before_action :check_staff_auth
+	def index
+	end
+
+	private
+
+	def check_staff_auth
+		if user_signed_in? && current_user.role >= 2
```
Comment (oliverochman) [inline app/controllers/employees_controller.rb]: where do you define the roles for the user?

File: app/views/devise/sessions/new.html.haml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+.form_container
+  .form_wrapper
+    %h2.text-center Log In
+    .form_container
+      = form_for(resource, as: resource_name, url: session_path(resource_name)) do |f|
```
Comment (oliverochman) [inline app/views/devise/sessions/new.html.haml]: have you modified this view in any way?

File: config/routes.rb [hunk 7]
```diff
@@ -3,4 +3,5 @@
 
 	root controller: :articles, action: :index
 	resources :articles, only: [:index, :show]
+	resources :employees, only: [:index]
```
Comment (oliverochman) [inline config/routes.rb]: I would think that the interesting thing for a Journalist to be redirected to when he logs in, is to a place where he can see his articles and create new ones.

So please add a namespace for a journalist or content management and add an articles controller for that namespace.

File: app/controllers/employees_controller.rb (hunk 5 above)
Comment (amandagibson) [reply app/controllers/employees_controller.rb]: it's in the stripe PR

File: app/views/devise/sessions/new.html.haml (hunk 6 above)
Comment (amandagibson) [reply app/views/devise/sessions/new.html.haml]: we added styling.

File: db/migrate/20190318083136_add_column_to_user.rb (hunk 4 above)
Comment (amandagibson) [reply db/migrate/20190318083136_add_column_to_user.rb]: done

File: config/routes.rb (hunk 7 above)
Comment (amandagibson) [reply config/routes.rb]: done

**Pull request merged**